### PR TITLE
Skip tests that check packages are up to date in CI

### PR DESCRIPTION
**Pull request by legoktm**
## Status

~~Ready for review~~ ~~in progress~~ Ready for review

## Description of Changes

In CI, the base templates are only updated once a day so it can easily be out of date.

## Testing

* [ ] Visual review
* [ ] test log shows tests are skipped

## Deployment

Any special considerations for deployment? n/a

### CODE REVIEW

**Comment by zenmonkeykstop** [comment]
Could we mark it xfail or to not be run in CI instead? The expectation when doing a dev install is that the system has been updated recently. I don't think we have to worry much about our packages at all, since they won't update that frequently.

IMO the optimal thing to do here would be to update the system templates automatically as part of the install process - which we can probably do as a shell callout with the Qubes updater now.

**Comment by legoktm** [comment]
> Could we mark it xfail or to not be run in CI instead?

Yes, both are possible. I don't think xfail is useful, and would rather go the "check if CI" route or remove it altogether.

> The expectation when doing a dev install is that the system has been updated recently.

I agree, however, I don't think it is useful to test for this property. As a developer, every time I do a `make test` run, I should also not need to run the updater. Or to put it another way, I don't think we should be verifying that the system under test is up to date, rather we should be verifying that the updater works and is able to keep the system up to date.

**Comment by rocodes** [comment]
IMO we do want overall system checks/tests, but probably run periodically against a prod/prodlike "real world conditions" system, not during CI or on dev machines (unless part of a release preflight CI somehow).  I think these tests are a relic from when `make dom0 test` was a `make sanity` check before publishing an rpm, as opposed to something meant to be scheduled and run periodically. 

A system test that runs the updater and then makes assertions about system state, separate from regular CI, is a good idea. (fwiw: kicksecure/whonix provides a `systemcheck` package: https://www.kicksecure.com/wiki/Systemcheck that gets at a similar goal, although primarily it's checking consistent system attributes plus their own metapackage versions).

tldr I'm fine with removing these tests from this part of CI since it's not a CI issue per se - we're more worried about the state of end user systems than anything else with these tests- but I would like to see what these tests represent being tested somewhere, eg as related to #939.

**Comment by legoktm** [comment]
> I think these tests are a relic from when `make dom0 test` was a `make sanity` check before publishing an rpm, as opposed to something meant to be scheduled and run periodically.

Gotcha, that makes sense.

> A system test that runs the updater and then makes assertions about system state, separate from regular CI, is a good idea.

👍🏾 

> tldr I'm fine with removing these tests from this part of CI since it's not a CI issue per se - we're more worried about the state of end user systems than anything else with these tests- but I would like to see what these tests represent being tested somewhere, eg as related to https://github.com/freedomofpress/securedrop-workstation/issues/939.

OK, once https://github.com/freedomofpress/securedrop-workstation-ci/pull/55 lands I'll update this to skip this test if run in CI. And when the updater work happens, we can work on better tests.

**Comment by rocodes** [comment]
[moving temporarily to `blocked` to reflect dependency on https://github.com/freedomofpress/securedrop-workstation-ci/pull/55]

**Comment by legoktm** [comment]
This is ready for review again :)

**Comment by legoktm** [comment]
From the log:
```
...
INFO:[2024-04-23-17:49:13:536805] test_all_fedora_vms_uptodate (test_vms_platform.SD_VM_Platform_Tests.test_all_fedora_vms_uptodate)
INFO:[2024-04-23-17:49:13:536835] Asserts that all Fedora-based templates, such as sys-net, have all ... skipped 'Skipping on CI'
INFO:[2024-04-23-17:49:13:537020] test_all_sd_vm_apt_sources (test_vms_platform.SD_VM_Platform_Tests.test_all_sd_vm_apt_sources)
INFO:[2024-04-23-17:49:13:537060] Test all VMs fpf apt source list iteratively. ... ok
INFO:[2024-04-23-17:49:13:537092] test_all_sd_vm_platforms (test_vms_platform.SD_VM_Platform_Tests.test_all_sd_vm_platforms)
INFO:[2024-04-23-17:49:13:537126] Test all VM platforms iteratively. ... ok
INFO:[2024-04-23-17:49:13:537160] test_all_sd_vms_uptodate (test_vms_platform.SD_VM_Platform_Tests.test_all_sd_vms_uptodate)
INFO:[2024-04-23-17:49:13:537191] Asserts that all VMs have all available apt packages at the latest ... skipped 'Skipping on CI'
...
```

**Review by zenmonkeykstop — APPROVED** [review]
LGTM based on visual review, CI results
